source/rock/middle/TypeDecl.ooc

```diff
@@ -196,7 +220,13 @@ TypeDecl: abstract class extends Declaration {
     getInterfaceDecls: func -> List<InterfaceImpl> { interfaceDecls }
 
     hasMeta?: func -> Bool {
-        !isMeta
+        if (debugCondition()) {
+            "hasMeta called, they want %s / %p back" printfln(toString(), template)
+        }
```

Comment:
All credit to @fasterthanlime, this is a copy pasted piece of code (from the cover template impl) and I'm not ashamed to say it :P